frontend/www/js/omegaup/contest/edit.ts

```diff
@@ -114,13 +114,76 @@ OmegaUp.on('ready', () => {
               })
               .catch(ui.apiError);
           },
+          'add-user': (contestants: string, contestant: string) => {
+            let users: string[] = [];
+            if (contestants !== '') {
+              users = contestants.split(',');
+            }
+            if (contestant !== '') {
+              users.push(contestant);
+            }
+            Promise.allSettled(
+              users.map((user: string) =>
+                api.Contest.addUser({
+                  contest_alias: payload.details.alias,
+                  usernameOrEmail: user.trim(),
+                }),
+              ),
+            )
+              .then((results) => {
+                let contestantsWithError: string[] = [];
+                results.forEach((result) => {
+                  if (result.status === 'rejected') {
+                    contestantsWithError.push(result.reason.userEmail);
+                  }
+                });
```

Comment:
nit:

```suggestion
                let contestantsWithError: string[] = results
                    .filter((result) => result.status === 'rejected')
                    .map((result) => result.reason.userEmail));
```

Reply:
Ok, según veo, el campo `reason.userEmail` nunca es retornado, así que esto genera un error de tipos. 
De hecho, me parece que nunca retornamos `rejected` tampoco, o al menos no consigo encontrarlo en el código del API.

Reply:
la presencia de `reason` y `rejected` es comportamiento cortesía de [`Promise.allSettled()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled#Return_value).

_pero_ eso aún no explica el origen de `userEmail`, que en efecto no está saliendo de ningún lado.

lo que podemos hacer es cambiar la invocación de `Promise.allSettled()` arriba por

```typescript
            Promise.allSettled(
              contestants.map((user: string) =>
                api.Contest.addUser({
                  contest_alias: payload.details.alias,
                  usernameOrEmail: user,
                })
                .catch(() => user),
              ),
            )
```

con eso, `result.reason` va a tener el nombre de usuario / correo, así que este snippet debería cambiarse por

```suggestion
                let contestantsWithError: string[] = results
                    .filter((result) => result.status === 'rejected')
                    .map((result) => result.reason));
```

también, con respecto al error de tipos, puedes compartir el error? hay que ver cómo arreglarlo.

Reply:
Listo!
Este es el error de tipos:
```bash
176:45 Property 'reason' does not exist on type 'PromiseSettledResult<string | ContestAddUserResponse>'.
  Property 'reason' does not exist on type 'PromiseFulfilledResult<string | ContestAddUserResponse>'.
```

Reply:
@lhchavez lo voy pusheando así, para que veas el error directamente acá.

Reply:
YA! gracias, ese error ya me dejó más claro el problema: 

```suggestion
                let contestantsWithError: string[] = results
                    .filter((result): item is PromiseRejectedResult => result.status === 'rejected')
                    .map((result) => result.reason));
```

lo que pasa es que `results` es una unión discriminada entre `PromiseFulfilledResult<ContestAddUserResponse>` y `PromiseRejectedResult`. la única manera de que TypeScript sepa cuál de los dos es el bueno es mediante esa anotación extra.

Reply:
Tampoco le gusta esa solución a Typescript.
Por cierto, cómo se llama ese tipo de asignación de tipos que acabas de proponer? La de `item is ...`?

Reply:
ah eso fue mi culpa, en vez de `item is PromiseRejectedResult` debió haber sido `result is PromiseRejectedResult` u_u

Eso se llama [type predicate](https://www.typescriptlang.org/docs/handbook/advanced-types.html#using-type-predicates).